### PR TITLE
Implemented GPIO to check processor present or not

### PR DESCRIPTION
**Pull request by ashok2ku**
Implemented GPIO to check CPU present or not and set the Dbus "present" property
Fixes: FWDEV-43511

### CODE REVIEW

File: src/cpu_info.cpp [hunk 1]
```diff
@@ -130,6 +141,7 @@ bool CpuInfo::connect_apml_get_family_model_step(uint8_t soc_num )
     uint32_t ext_model;
     int core_id = 0;
     uint16_t freq;
+    uint16_t cpuvalue;
```
Comment (supven01) [inline src/cpu_info.cpp:144]: nit: misnomer cpuvalue. Use better variable name "cpuPresence" or something similar.

File: src/cpu_info.cpp [hunk 2]
```diff
@@ -150,6 +162,28 @@ bool CpuInfo::connect_apml_get_family_model_step(uint8_t soc_num )
...
+         //set false -Absent if GPIO value is high -default is true
+	 cpuvalue = getGPIOValue(P0_Present);
+         if (cpuvalue == 1)
+         {
+            set_cpu_bool_value(soc_num, false, DBUS_Present, CPU_INTERFACE);
+	    sd_journal_print(LOG_INFO, "Warning : %d CPU is absent \n", soc_num);
+            return false;
+         }
+      }
+      else if (soc_num == 1)
```
Comment (supven01) [inline src/cpu_info.cpp:176]: do we need else if here? 
it should be just "if"

**Review by supven01 — CHANGES_REQUESTED** [review]

File: src/cpu_info.cpp (hunk 1 above)
Comment (ashok2ku) [reply src/cpu_info.cpp]: fixed

File: src/cpu_info.cpp (hunk 2 above)
Comment (ashok2ku) [reply src/cpu_info.cpp]: fixed

File: src/cpu_info.cpp [hunk 3]
```diff
@@ -150,6 +162,28 @@ bool CpuInfo::connect_apml_get_family_model_step(uint8_t soc_num )
         }
       }//end of retry
 
+      if (soc_num == 0)
```
[the remark this answers is not in the document]
Comment (ashok2ku) [reply src/cpu_info.cpp]: fixed

**Review by supven01 — APPROVED** [review]

**Pull request merged**